src/modules/adaptors/Morpho/MorphoAaveV2DebtTokenAdaptor.sol

```diff
@@ -0,0 +1,147 @@
+        IAaveToken aToken = abi.decode(adaptorData, (IAaveToken));
+        return ERC20(aToken.UNDERLYING_ASSET_ADDRESS());
+    }
+
+    /**
+     * @notice This adaptor reports values in terms of debt.
+     */
+    function isDebt() public pure override returns (bool) {
+        return true;
+    }
+
+    //============================================ Strategist Functions ===========================================
+
+    /**
+     * @notice Allows strategists to borrow assets from Aave.
+     * @notice `aToken` must be the aToken not the debtToken.
+     * @param aToken the aToken to borrow on Aave
+     * @param amountToBorrow the amount of `aTokenToBorrow` to borrow on Morpho.
+     */
+    function borrowFromAaveV2Morpho(address aToken, uint256 amountToBorrow) public {
+        // Check that debt position is properly set up to be tracked in the Cellar.
+        bytes32 positionHash = keccak256(abi.encode(identifier(), true, abi.encode(aToken)));
+        uint32 positionId = Cellar(address(this)).registry().getPositionHashToPositionId(positionHash);
+        if (!Cellar(address(this)).isPositionUsed(positionId))
+            revert MorphoAaveV2DebtTokenAdaptor__DebtPositionsMustBeTracked(aToken);
+
+        // Borrow from morpho.
+        morpho().borrow(aToken, amountToBorrow);
+    }
+
+    /**
+     * @notice Allows strategists to repay loan debt on Aave.
+     * @param aToken the aToken you want to repay.
+     * @param amountToRepay the amount of `tokenToRepay` to repay with.
+     */
+    function repayAaveV2MorphoDebt(IAaveToken aToken, uint256 amountToRepay) public {
+        ERC20 underlying = ERC20(aToken.UNDERLYING_ASSET_ADDRESS());
+        if (amountToRepay == type(uint256).max) {
+            uint256 availableUnderlying = underlying.balanceOf(address(this));
+            uint256 debt = _balanceOfInUnderlying(address(aToken), address(this));
+            amountToRepay = availableUnderlying > debt ? debt : availableUnderlying;
+        }
+        underlying.safeApprove(address(morpho()), amountToRepay);
+        morpho().repay(address(aToken), amountToRepay);
+
+        // Zero out approvals if necessary.
+        _revokeExternalApproval(underlying, address(morpho()));
+    }
+
+    /**
+     * @notice Returns the balance in underlying of debt owed.
+     * @param poolToken the Aave V2 a Token user has debt in
+     * @param user the address of the user to query their debt balance of.
+     */
+    function _balanceOfInUnderlying(address poolToken, address user) internal view returns (uint256) {
+        (uint256 inP2P, uint256 onPool) = morpho().borrowBalanceInOf(poolToken, user);
+
+        uint256 balanceInUnderlying;
+        if (inP2P > 0) balanceInUnderlying = inP2P.mulDivDown(morpho().p2pBorrowIndex(poolToken), 1e27);
+        if (onPool > 0) balanceInUnderlying += onPool.mulDivDown(morpho().poolIndexes(poolToken).poolBorrowIndex, 1e27);
```

Comment:
Bear in mind that these indexes may be outdated (they are updated at each interaction with the protocol and thus are not guaranteed to be up-to-date). If you want to access the true balance in underlying, you need to use the indexes calculated by the Lens (or calculate them yourself)